src/client/eppo-client.spec.ts

```diff
@@ -88,1231 +106,601 @@ describe("EppoClient E2E test", () => {
-      subjectAttributes?: Record<string, any>;
-    }[],
-    experiment: string,
-    valueTestType: ValueTestType = ValueTestType.StringType
-  ): (EppoValue | null)[] {
-    return subjectsWithAttributes.map((subject) => {
-      switch (valueTestType) {
-        case ValueTestType.BoolType: {
-          const ba = globalClient.getBoolAssignment(
-            subject.subjectKey,
-            experiment,
-            subject.subjectAttributes,
-            undefined,
-            true
-          );
-          if (ba === null) return null;
-          return EppoValue.Bool(ba);
-        }
-        case ValueTestType.NumericType: {
-          const na = globalClient.getNumericAssignment(
-            subject.subjectKey,
-            experiment,
-            subject.subjectAttributes,
-            undefined,
-            true
-          );
-          if (na === null) return null;
-          return EppoValue.Numeric(na);
-        }
-        case ValueTestType.StringType: {
-          const sa = globalClient.getStringAssignment(
-            subject.subjectKey,
-            experiment,
-            subject.subjectAttributes,
-            undefined,
-            true
-          );
-          if (sa === null) return null;
-          return EppoValue.String(sa);
-        }
-        case ValueTestType.JSONType: {
-          const sa = globalClient.getJSONStringAssignment(
-            subject.subjectKey,
-            experiment,
-            subject.subjectAttributes,
-            undefined,
-            true
-          );
-          const oa = globalClient.getParsedJSONAssignment(
-            subject.subjectKey,
-            experiment,
-            subject.subjectAttributes,
-            undefined,
-            true
-          );
-          if (oa == null || sa === null) return null;
-          return EppoValue.JSON(sa, oa);
-        }
-      }
+    afterAll(() => {
+      mock.teardown();
     });
-  }
-});
 
-describe("Eppo Client constructed with configuration request parameters", () => {
-  let client: EppoClient;
-  let storage: IConfigurationStore;
-  let requestConfiguration: ExperimentConfigurationRequestParameters;
-  let mockServerResponseFunc: (res: MockResponse) => MockResponse;
-
-  const racBody = JSON.stringify(readMockRacResponse(MOCK_RAC_RESPONSE_FILE));
-  const flagKey = "randomization_algo";
-  const subjectForGreenVariation = "subject-identiferA";
-
-  const maxRetryDelay = POLL_INTERVAL_MS * POLL_JITTER_PCT;
-
-  beforeAll(() => {
-    mock.setup();
-    mock.get(/randomized_assignment\/v3\/config*/, (_req, res) => {
-      return mockServerResponseFunc(res);
+    it('Fetches initial configuration with parameters in constructor', async () => {
+      client = new EppoClient(evaluator, storage, requestConfiguration);
+      client.setIsGracefulFailureMode(false);
+      // no configuration loaded
+      let variation = client.getNumericAssignment(subject, flagKey, 0.0);
```

Comment:
this _may_ cover that uninitialized condition I was concerned about (but we should double check)